packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -39,46 +45,72 @@ export class SummarizeHeuristicData implements ISummarizeHeuristicData {
             refSequenceNumber: refSequenceNumber ?? this.lastOpSequenceNumber,
             summaryTime: Date.now(),
         };
+
+        this.numSystemOpsBefore = this.numSystemOps;
+        this.numSystemOps = 0;
+
+        this.numNonSystemOpsBefore = this.numNonSystemOps;
+        this.numNonSystemOps = 0;
     }
 
     public markLastAttemptAsSuccessful() {
         this._lastSuccessfulSummary = { ...this.lastAttempt };
     }
+
+    public adjustOnFailure(): void {
+        this.numSystemOps += this.numSystemOpsBefore;
+        this.numNonSystemOps += this.numNonSystemOpsBefore;
+    }
 }
```

Comment:
I wonder if it's better to do something like that:
```ts
recordAttempt() {
        this.numSystemOpsBefore = this.numSystemOps;
        this.numNonSystemOpsBefore = this.numNonSystemOps;
}

public markLastAttemptAsSuccessful() {
      this.numSystemOps -= this.numSystemOpsBefore;
      this.numSystemOpsBefore = 0;
      this.numNonSystemOps -= this.numNonSystemOpsBefore;
      this.numNonSystemOpsBefore = 0;
}
```

Feels a bit cleaner to me